content/en/altinity-kb-queries-and-syntax/trace_log.md

```diff
@@ -37,56 +37,78 @@ pip3 install -r requirements.txt
 ```
 
 For debugging particular query:
-```
+1. In the clickhouse-client, set your desired profiler settings:
+```sql
 clickhouse-client 
 
 SET query_profiler_cpu_time_period_ns=1000000; -- 1000 times per 'cpu' sec
--- or SET query_profiler_real_time_period_ns=2000000; -- 500 times per 'real' sec.
--- or SET memory_profiler_sample_probability=0.1; -- to debug the memory allocations
-
+-- or
+SET query_profiler_real_time_period_ns=2000000; -- 500 times per 'real' sec.
+-- or
+SET memory_profiler_sample_probability=0.1; -- to debug the memory allocations
+```
+2. Execute your query:
+```sql
 SELECT ... <your select>
-
+```
+3. Flush the logs:
+```sql
 SYSTEM FLUSH LOGS;
-
--- get the query_id from the clickhouse-client output or from system.query_log (also pay attention on query_id vs initial_query_id for distributed queries).
 ```
+4. Retrieve the query_id from either the clickhouse-client output or from system.query_log (note the difference between query_id and initial_query_id for distributed queries).
+5. Process the stack traces:
 
-Now let's process that: 
-```
+```bash
 python3 main.py &  # start the proxy in background
 python3 main.py --query-id 908952ee-71a8-48a4-84d5-f4db92d45a5d # process the stacktraces
 fg # get the proxy from background 
 Ctrl + C  # stop it.
 ```
-
-To access ClickHouse with other username / password etc. - see the sources of https://github.com/laplab/clickhouse-speedscope/blob/master/main.py
-
+For additional configuration options, refer to the source of the tool: clickhouse-speedscope https://github.com/laplab/clickhouse-speedscope/blob/master/main.py
 
 ### clickhouse-flamegraph
 
-Installation & usage instructions: https://github.com/Slach/clickhouse-flamegraph
+For more advanced visualizations, clickhouse-flamegraph can be used. Installation and usage instructions are available on its GitHub page https://github.com/Slach/clickhouse-flamegraph
 
-### pure flamegraph.pl examples
+Example for Generating a Flamegraph:
 
-```
+1. Install FlameGraph:
+```bash
 git clone https://github.com/brendangregg/FlameGraph /opt/flamegraph
-
+```
+2. Generate a flamegraph from the system.trace_log:
```

Comment:
this is not the second step with clickhouse-flamegraph, this is different approach